vktrace/vktrace_layer/vktrace_lib_trim.cpp

```diff
@@ -854,7 +854,8 @@ void generateMapUnmap(bool makeCalls, VkDevice device, VkDeviceMemory memory, Vk
         // unmap for it.
         vktrace_delete_trace_packet(&pMapMemory);
     } else {
-        *ppMapMemoryPacket = pMapMemory;
+        *ppMapMemoryPacket = trim::copy_packet(pMapMemory);
+        vktrace_delete_trace_packet(&pMapMemory);
```

Comment:
This change seems like it is unrelated to the gist of this pull request. If it is unrelated, could you make this change in a separate pull request? Or maybe I'm wrong?

Reply:
You are correct.  It is not fixing the issue of "unlock uninitialized mutex lock" but a fix to "lock on a locked mutex lock".  I forgot to update the description of the pull request when pushing the later two commits.  Shall I update the description of this pull request or create a new one?

I think it is better to have both fixes being merged together since both of them are needed to fix the mutex lock problem.